src/main/java/io/cryostat/net/reports/RemoteReportGenerator.java

```diff
@@ -80,16 +81,27 @@ class RemoteReportGenerator extends AbstractReportGeneratorService {
 
     @Override
     @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
-    public CompletableFuture<Path> exec(Path recording, Path destination) {
+    public CompletableFuture<Path> exec(Path recording, Path destination, String filter) {
         String reportGenerator = env.getEnv(Variables.REPORT_GENERATOR_ENV);
+        System.out.println("OR IS IT HERE IN REMOTE?");
         logger.info("POSTing {} to {}", recording, reportGenerator);
+        System.out.println(String.format("IN REMOTE: THIS IS THE FILTER {%s}", filter));
         var form =
-                MultipartForm.create()
-                        .binaryFileUpload(
-                                "file",
-                                recording.getFileName().toString(),
-                                recording.toAbsolutePath().toString(),
-                                HttpMimeType.OCTET_STREAM.mime());
+                StringUtils.isNotBlank(filter)
+                        ? MultipartForm.create()
```

Comment:
This seems like an awkward use of the ternary operator. Can't the form be created, the `.binaryFileUpload` applied regardless, and then just the `.attribute()` performed conditionally?